### PR TITLE
UTC: support debug output from LDist

### PR DESCRIPTION
**Pull request by artagnon**
Tweak the LoopDistribute debug output to be prefixed with "LDist: ", get it to be stable, and extend update_analyze_test_checks.py trivially to support this output.

### CODE REVIEW

**Comment by artagnon** [comment]
Gentle ping. I'm planning to make improvements to LoopDistribute, and this patch is a prerequisite.

**Comment by artagnon** [comment]
Rebase and ping.

**Comment by artagnon** [comment]
Gentle ping. This patch should be straightforward.

File: llvm/test/tools/UpdateTestChecks/update_analyze_test_checks/loop-distribute.test [hunk 1]
```diff
@@ -0,0 +1,6 @@
+## Basic test checking that update_analyze_test_checks.py works correctly
+# RUN: cp -f %S/Inputs/loop-distribute.ll %t.ll && %update_analyze_test_checks %t.ll
+# RUN: diff -u %t.ll %S/Inputs/loop-distribute.ll.expected
+## Check that running the script again does not change the result:
+# RUN: %update_analyze_test_checks %t.ll
+# RUN: diff -u %t.ll %S/Inputs/loop-distribute.ll.expected
```
Comment (nikic) [inline llvm/test/tools/UpdateTestChecks/update_analyze_test_checks/loop-distribute.test:6]: You probably need to require asserts, otherwise debug-only is not available.

File: llvm/utils/UpdateTestChecks/common.py [hunk 2]
```diff
@@ -1635,8 +1636,9 @@ def generalize_check_lines(
         regexp = ginfo.get_regexp()
 
     multiple_braces_re = re.compile(r"({{+)|(}}+)")
+
     def escape_braces(match_obj):
-        return '{{' + re.escape(match_obj.group(0)) + '}}'
+        return "{{" + re.escape(match_obj.group(0)) + "}}"
```
Comment (nikic) [inline llvm/utils/UpdateTestChecks/common.py:1641]: Unrelated changes?

File: llvm/lib/Transforms/Scalar/LoopDistribute.cpp [hunk 3]
```diff
@@ -120,7 +120,7 @@ namespace {
 /// Maintains the set of instructions of the loop for a partition before
 /// cloning.  After cloning, it hosts the new loop.
 class InstPartition {
-  using InstructionSet = SmallPtrSet<Instruction *, 8>;
+  using InstructionSet = SetVector<Instruction *>;
```
Comment (nikic) [inline llvm/lib/Transforms/Scalar/LoopDistribute.cpp:123]: Does SmallSetVector make sense?

File: llvm/utils/UpdateTestChecks/common.py (hunk 2 above)
Comment (artagnon) [reply llvm/utils/UpdateTestChecks/common.py:1641]: Unfortunately, `black` doesn't have a way to format only the diff (as far as I know), and the CI would fail without this change.

File: llvm/lib/Transforms/Scalar/LoopDistribute.cpp [hunk 4]
```diff
@@ -231,17 +231,16 @@ class InstPartition {
...
+  void print(raw_ostream &OS) const {
+    OS << (DepCycle ? " (cycle)\n" : "\n");
     for (auto *I : Set)
       // Prefix with the block name.
-      dbgs() << "  " << I->getParent()->getName() << ":" << *I << "\n";
+      OS << "  " << I->getParent()->getName() << ":" << *I << "\n";
   }
 
-  void printBlocks() const {
+  void printBlocks(raw_ostream &OS) const {
```
Comment (fhahn) [inline llvm/lib/Transforms/Scalar/LoopDistribute.cpp:241]: Is adding the output stream as arg needed? AFAICT it will only be `dbgs()` with this patch?

**Review by fhahn — COMMENTED** [review]
Would be good to also mentioned that this updates LDist in the patch, or split up the patches into 2, one updating the debug output and one for UTC

File: llvm/lib/Transforms/Scalar/LoopDistribute.cpp (hunk 4 above)
Comment (artagnon) [reply llvm/lib/Transforms/Scalar/LoopDistribute.cpp:241]: The `print()` function on line 530 takes an output stream, so I thought I should change the other `print` function and `printBlocks` to take an output stream as well, for uniformity. Besides, this is also the convention used in other `dump()` functions in the tree.

Comment (artagnon) [reply llvm/lib/Transforms/Scalar/LoopDistribute.cpp:241]: I additionally investigated stripping the `raw_ostream` argument everywhere, but this is not possible due to `operator<<`.

**Review by nikic — APPROVED** [review]
LGTM

**Review by fhahn — APPROVED** [review]
Thanks for clarification re print() already taking OS

LGTM
